docs/federate.md

```diff
@@ -0,0 +1,135 @@
+which is valid for your ``server_name``.
+
+Once you have completed the steps necessary to federate, you should be able to
+ [connect to your server with a client](../README.rst#registering-a-new-user-from-a-client).
+and then join a room via federation. (A good place to start is ``#synapse:matrix.org``
+ - a room for Synapse admins.)
+
+
+## Delegation
+
+For a more flexible configuration, you can have ``server_name``
+resources (ie: ``@user:example.com``) served by a different host and
+port (ie: ``synapse.example.com:443``). There are two ways to do this:
+
+- adding a ``/.well-known/matrix/server`` URL served on ``https://example.com``.
+- adding a DNS ``SRV`` record in the DNS zone of domain
+  ``example.com``.
+
+For both methods let's say you want to run your server at
+``synapse.example.com`` on port ``443`` (instead of ``8448``), but you
+want to have your Matrix user-ids look like ``@user:example.com``.
+
+Without configuring delegation, the matrix federation will
+expect to find your resources via ``example.com:8448``. The
+following methods allow you to provide a different server and port for
+``*:example.com`` resources.
+
+### .well-known delegation
+
+To use this method, you need to be able to alter the
+``server_name`` 's https server to serve the ``/.well-known/matrix/server``
+URL. Having an active server (with a valid ``SSL`` certificate) serving your
+``server_name`` domain is out of the scope of this documentation.
+
+The URL ``https://<server_name>/.well-known/matrix/server`` should
+return a JSON structure containing the key ``m.server`` like so:
+
+    {
+	    "m.server": "<synapse.server.name>:<yourport>"
+    }
+
+In our example, this would mean that URL ``https://example.com/.well-known/matrix/server``
+should return:
+
+    {
+	    "m.server": "synapse.example.com:443"
+    }
+
+Note, specifying a port is optional and will default to 443.
+
+Most installations will not need to configure .well-known. However it can  be
+useful in cases where the admin is hosting on behalf of someone else and
+therefore cannot gain access to the necessary certificate. With .well-known,
+federation servers will contact the given hostname's IP and
+will check for a valid SSL certificate on the same delegated hostname (in our
+example: ``synapse.example.com``).
+
+.well-known support first appeared in Synapse v0.99.0, to federate with older
+servers you may need to additionally configure SRV delegation. Alternatively,
+encourage the server admin in question to upgrade :)
+
+### DNS SRV delegation
+
+
+To use this method, you need to have write access to your
+``server_name`` 's domain zone DNS records (in our example it would be
+``example.com`` DNS zone).
+
+This method requires the target server to provide a
+valid SSL certificate identifying it as the original ``server_name``
+domain zone. So the delegate domain name is only used to resolve a possible
+different IP/Port combination to find your server. You must use the other
```

Comment:
I'm not sure this sentence adds much either, but if you think it's worth keeping, suggest rewording as "You should use [.well-known delegation][link] if this isn't what you want.".

Reply:
Removed